integrations/extensions/starter-kits/search-with-milvus/search-with-watsonx-data-milvus.md

```diff
@@ -0,0 +1,109 @@
+
+### Option 2: Ingest data using custom code
+Here is a sample code to ingest documents into Milvus: [../search-with-custom-service/examples/index-with-milvus.py](../search-with-custom-service/examples/index-with-milvus.py). To run it, 
+1. Install dependencies:
+   ```bash
+   python3 -m pip install pymilvus langchain langchain-milvus langchain-ibm ibm-watsonx-ai
+   ```
+2. Create environment variables for Milvus credentials
+  ```bash
+  export MILVUS_HOST = 'Your Milvus GRPC host'
+  export MILVUS_PORT = 'Your Milvus GRPC port'
+  export MILVUS_USER = 'ibmlhapikey' // your Milvus username
+  export MILVUS_PASSWORD = 'Your watsonx.data API key'
+
+  export MILVUS_COLLECTION_NAME = 'Your Milvus collection name' // It can be anything
+  export WATSONX_AI_APIKEY = 'Your watsonx.ai API key' // watsonx.ai embeddings model is used to create vectors
+  export WATSONX_AI_PROJECT_ID = 'Your watsonx.ai project ID' // watsonx.ai project ID is required to access the models
+  ```
+3. Update the `SOURCE_FILE_NAMES`, `SOURCE_URLS`, and `SOURCE_TITLES` variables at the begining of the script to your file names, urls, and titles respectively.
+4. Run the script
+   ```bash
+   python3 index-with-milvus.py
+   ```
+
+## Step 3: Connect to watsonx Assistant
+There are two ways to connect your Milvus vector database to watsonx Assistant for search and conversational search, either through the built-in Milvus search integration or using the Custom Service search.
+
+### Option 1: Use the built-in Milvus search integration
+Please go to the [watsonx Assistant product docs](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-search-overview) to find more details about how to set up the built-in Milvus search integration.
+
+### Option 2: Use the Custom Service search
+Custom Service search provides a more advanced and flexible way to connect your Milvus vector database to watsonx Assistant. It requires you to set up a custom server that must be reachable by your watsonx Assistant instance. 
+Then you need to configure the custom server by providing your server credentials and metadata. 
+
+#### Set up a custom server
+Here is an example server code based on Python Flask: [../search-with-custom-search/examples/example-milvus-server.py](../search-with-custom-service/examples/example-simple-server.py). This example code requires setting a list of environment variables, including `MILVUS_HOST`, `MILVUS_PORT`, `MILVUS_USER`, `MILVUS_PASSWORD`, `WATSONX_AI_APIKEY`, `WATSONX_AI_PROJECT_ID`. 
+
+You need to deploy the code as a service on either Cloud or Cloud Pak for Data and make it reachable by your watsonx Assistant instance. IBM Code Engine is recommended to host the code. See [Code Engine docs](https://cloud.ibm.com/docs/codeengine?topic=codeengine-getting-started) for more details.
```

Comment:
Do you think people who use Cloud Pak for Data would use IBM Code Engine?  It's only available for SaaS/IBM Cloud.
Maybe could just reword to say: "If you are using Cloud, IBM Code Engine is recommended to host the code. "  

For Cloud Pak for Data, they can figure out where to run a service accessible to them. You wouldn't need to recommend something specifically.